internal/manager/run.go

```diff
@@ -197,7 +197,16 @@ func Run(ctx context.Context, c *Config, diagnostic util.ConfigDumpDiagnostic, d
 		if err != nil {
 			setupLog.Error(err, "failed creating konnect client, skipping running NodeAgent")
 		} else {
-			hostname, _ := os.Hostname()
+			var hostname string
+			podName := os.Getenv("POD_NAME")
+			podNs := os.Getenv("POD_NAMESPACE")
+			if podName == "" || podNs == "" {
+				setupLog.Info("pod namespace or pod name not found, fallback to use hostname as node name in konnect")
+				hostname, _ = os.Hostname()
+			} else {
+				hostname = podNs + "/" + podName
```

Comment:
If `podNs` ends up empty then `hostname` becomes: `"/" + podName` and vice versa for empty `podName`. Is that intended?

Reply:
I think either being empty will make the hostname be `os.Hostname()`.